napari/__init__.py

```diff
@@ -38,6 +38,12 @@
 
 del stats
 
+
+import os
+
+os.environ['SPARSE_AUTO_DENSIFY'] = '1'
```

Comment:
```suggestion
os.environ.setdefault('SPARSE_AUTO_DENSIFY', '1')
```

This will only set it if it is not already set. I'm ok with us adding this setting in general, I'm less ok with overriding someone's explicit setting of 0!

Other than that I'm ok with having this in the init.